recipes/graphene/all/conanfile.py

```diff
@@ -52,64 +62,74 @@ def configure(self):
             self.options["glib"].shared = True
 
     def validate(self):
-        if self.options.shared and self.options.with_glib and not self.options["glib"].shared:
-            raise ConanInvalidConfiguration(
-                "Linking a shared library against static glib can cause unexpected behaviour."
-            )
+        if self.info.options.with_glib:
+            glib_is_shared = self.dependencies.direct_host["glib"].options.shared
```

Comment:
any special requirement to use `direct_host` here? Is not `self.dependencies["glib"]` enough?

Reply:
Didn't work last time I tried it - recent change maybe? Fixed now.